beacon-chain/blockchain/service.go

```diff
@@ -232,23 +232,26 @@ func (c *ChainService) updateHead(slotInterval <-chan time.Time) {
 			}
 
 			log.Info("Applying fork choice rule")
-			aState := c.chain.ActiveState()
+
+			parentBlock, err := c.chain.getBlock(block.ParentHash())
+			if err != nil {
+				log.Errorf("Failed to get parent of block 0x%x", h)
+				continue
+			}
+
 			cState := c.chain.CrystallizedState()
-			isTransition := cState.IsCycleTransition(c.currentSlot - 1)
+			aState := c.chain.ActiveState()
+			var stateTransitioned bool
 
-			if isTransition {
+			for cState.IsCycleTransition(parentBlock.SlotNumber()) {
 				cState, err = cState.NewStateRecalculations(aState, block)
 				if err != nil {
 					log.Errorf("Initialize new cycle transition failed: %v", err)
 					continue
 				}
+				stateTransitioned = true
```

Comment:
This variable and `isTransition` is redundant. I'd suggest just sticking to `isTransition`.

Reply:
whoops, ignore this!